samples/hello-cube.html

```diff
@@ -0,0 +1,336 @@
+
+let device, context, verticesBuffer, mappedTransformArray, uniformsBindGroup, pipeline, renderPassDescriptor, queue;
+let projectionMatrix = mat4.create();
+
+const vertexSize = 4 * 8;
+const verticesSize = vertexSize * 36;
+const colorOffset = 4 * 4;
+function populateVerticesArray(array) {
+    const vertexData = [
+        // float4 position, float4 color
+        1, -1, 1, 1, 1, 0, 1, 1,
+        -1, -1, 1, 1, 0, 0, 1, 1,
+        -1, -1, -1, 1, 0, 0, 0, 1,
+        1, -1, -1, 1, 1, 0, 0, 1,
+        1, -1, 1, 1, 1, 0, 1, 1,
+        -1, -1, -1, 1, 0, 0, 0, 1,
+
+        1, 1, 1, 1, 1, 1, 1, 1,
+        1, -1, 1, 1, 1, 0, 1, 1,
+        1, -1, -1, 1, 1, 0, 0, 1,
+        1, 1, -1, 1, 1, 1, 0, 1,
+        1, 1, 1, 1, 1, 1, 1, 1,
+        1, -1, -1, 1, 1, 0, 0, 1,
+
+        -1, 1, 1, 1, 0, 1, 1, 1,
+        1, 1, 1, 1, 1, 1, 1, 1,
+        1, 1, -1, 1, 1, 1, 0, 1,
+        -1, 1, -1, 1, 0, 1, 0, 1,
+        -1, 1, 1, 1, 0, 1, 1, 1,
+        1, 1, -1, 1, 1, 1, 0, 1,
+
+        -1, -1, 1, 1, 0, 0, 1, 1,
+        -1, 1, 1, 1, 0, 1, 1, 1,
+        -1, 1, -1, 1, 0, 1, 0, 1,
+        -1, -1, -1, 1, 0, 0, 0, 1,
+        -1, -1, 1, 1, 0, 0, 1, 1,
+        -1, 1, -1, 1, 0, 1, 0, 1,
+
+        1, 1, 1, 1, 1, 1, 1, 1,
+        -1, 1, 1, 1, 0, 1, 1, 1,
+        -1, -1, 1, 1, 0, 0, 1, 1,
+        -1, -1, 1, 1, 0, 0, 1, 1,
+        1, -1, 1, 1, 1, 0, 1, 1,
+        1, 1, 1, 1, 1, 1, 1, 1,
+
+        1, -1, -1, 1, 1, 0, 0, 1,
+        -1, -1, -1, 1, 0, 0, 0, 1,
+        -1, 1, -1, 1, 0, 1, 0, 1,
+        1, 1, -1, 1, 1, 1, 0, 1,
+        1, -1, -1, 1, 1, 0, 0, 1,
+        -1, 1, -1, 1, 0, 1, 0, 1,
+    ];
+
+    for (let i = 0; i < vertexData.length; ++i) {
+        array[i] = vertexData[i];
+    }
+}
+
+const transformSize = 4 * 16;
+function populateTransformArray(array) {
+    let viewMatrix = mat4.create();
+    mat4.translate(viewMatrix, viewMatrix, vec3.fromValues(0, 0, -5));
+    let now = Date.now() / 1000;
+    mat4.rotate(viewMatrix, viewMatrix, 1, vec3.fromValues(Math.sin(now), Math.cos(now), 0));
+
+    let modelViewProjectionMatrix = mat4.create();
+    mat4.multiply(modelViewProjectionMatrix, projectionMatrix, viewMatrix);
+
+    for (let i = 0; i < 16; i++) {
+        array[i] = modelViewProjectionMatrix[i];
+    }
+}
+
+async function init() {
+    const adapter = await webgpu.requestAdapter({}); // Current (unifinished) implementation of requestAdapter requires this argument.
+    device = adapter.createDevice();
+
+    const canvas = document.querySelector('canvas');
+
+    const aspect = Math.abs(canvas.width / canvas.height);
+    mat4.perspective(projectionMatrix, (2 * Math.PI) / 5, aspect, 1, 100.0);
+
+    context = canvas.getContext('webgpu');
+
+    const swapChainDescriptor = { 
+        device: device, 
+        format:"B8G8R8A8Unorm", 
+        width: canvas.width, 
+        height: canvas.height 
+    };
+    context.configure(swapChainDescriptor);
+
+    // WebKit WebGPU accepts only MSL for now.
```

Comment:
Let's be more explicit and call out that we didn't decide which language WebGPU ingests yet.

Reply:
+1, I'd it would be nice to have a big scary warning around the shader before we merge it in. It should also specify that the API will *definitely not* accept Metal SL.